### ISSUE TITLE
"Lookup mode" - use the extension to find a password without browsing to the site

### ISSUE THREAD

**Issue by ttyridal** [ttyridal/masterpassword-firefox#72]
moving the discussion here after a rather long discussion in PR #68 

The future branch is blurring the line between stored password setting(s) that are related to the currently visited site (active url) and all (for some value of all) sites the user could want a password for. Typically to use outside of the browser.

short summary of discussion
* looking up a non-related password would wrongfully add the currently visited url to that password configuration 
* It might confuse users, and in particular(?) reduce the motivation to entering good site names. 
* loosing or polluting the link between the site name (password settings) and visited url will make it hard for the extension to be context-aware: propose the correct site name when opened on website.
* The lookup feature is desired :)

need solution for
* know when to record the active url as "related" for a password setting
* Passwords that are not stored because the user only ever used the default settings for that site. (is this really a problem? could just save with url = sitename the first time that password is looked up).. Does it need to save at all? If it's saved, it could auto-complete, so less typing.

**Comment by ttyridal** [comment]
@jmorgannz. (moving the discussion here)

>>I see.. so I'm about to make the extension too limit-less and non-tech-user-friendly 😮 can't have any of that 🤣🤣

>Hah. Hope you don't take any of my comments the wrong way. It's your baby! Only sharing my experience with a couple of users.

Not at all, and it's valuable input! I guess most people don't care too much, so make it simple for them, but give them a gentle nudge in the right direction. And then again. In my mind, the whole master password idea is more likely to appeal to the power user, so cater for them first.

> By definition the only URL's available in this mode would be stored, right - unless a user manually typed one in.
If they did; wouldn't that be a pretty good hint to save it?

Correct, you would have to type it out - and that would indeed be a very good opportunity to save it for auto-completion and putting it on the list. -  setting url = sitename would be the only logical choice. It's what would have happened if you pressed some hypothetical save button with only default values filled anyway.

> * Somehow use browser history to glean a list of domains for the same [getting relevant non-saved site names] purpose?

Ohh.. I like that!  if not for populating the list, at least it could be input for auto completion.   It does need another permission. I guess asking for less is always better when someone considers installing the extension. Don't know how important this is for people  (non-power-users probably don't care at all :))

**Comment by jmorgannz** [comment]
Could have sworn I replied to this.  

Don't really have anything more to add after the summarising and shift to this issue request.  
I'd like to have a bash on implementing some of the bits and pieces we talked about as little nibble PR's - but I've been waylaid with other life things for the moment - and TBH the PR's I did on old masterpassword-chrome are working for me so I haven't had the annoyance motivate me to go make code changes!
